server/src/main/java/keywhiz/service/daos/AclDAO.java

```diff
@@ -263,7 +266,17 @@ public void findAndEvictClient(long clientId, long groupId) {
               secretContentMapper.tryToReadMapFromMetadata(row.getValue(SECRETS_CONTENT.METADATA)),
               series.type().orElse(null),
               series.generationOptions(),
-              row.getValue(SECRETS_CONTENT.EXPIRY));
+              row.getValue(SECRETS_CONTENT.EXPIRY),
+              Collections.emptyList()
+          );
+          LazyString lazyString = () -> "";
+          Set<Group> groups = getGroupsFor(
+              new Secret(secret.id(), secret.name(), secret.description(), lazyString, secret.createdAt(),
```

Comment:
Instead of doing this here (and below), we should just have an API somewhere that fetches all the groups based on a secret name. If you look at `getGroupsFor`, it just reads the name of the secret and doesn't care about anything else. It was probably typed to take a `Secret` just for convenience.